collector/exchange.go

```diff
@@ -9,20 +9,34 @@ import (
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/common/log"
-	"gopkg.in/alecthomas/kingpin.v2"
 )
 
+var exchangeList = Config{
+	Name:     "collectors.exchange.list",
+	HelpText: "List the collectors along with their perflib object name/ids",
+	Default:  "false",
+}
+var exchangeEnabled = Config{
+	Name:     "collectors.exchange.enabled",
+	HelpText: "Comma-separated list of collectors to use. Defaults to all, if not specified.",
+	Default:  "",
+}
```

Comment:
Exposing the configs this way makes me nervous:

* It isn't strongly typed, so importers don't know if they got a flag name wrong until runtime 
* All usage of kingpin has to be swapped out with this internal config lib, which makes it harder to justify upstream (it has the appearance of a more disruptive change rather than just shifting logic around) 

I'd strongly recommend avoiding doing things this way and to use [Prometheus' service discovery mechanism](https://github.com/prometheus/prometheus/blob/82ffdae1f3da6ace0fd61fddc766f88677a70e9d/discovery/kubernetes/kubernetes.go#L113-L128) as inspiration:

1. Create public per-collector config structs with public fields for each tunable 
2. Add a `RegisterFlags` method (or similar) that binds the values of the field to a flag. 
3. Add a `Build` or `NewCollector` method that creates the collector, passing the config to it as an argument.

Reply:
Each config needs to pass 3 things: name, default and helptext. Will try to see how to make that work with using the mechanism linked above. Also isnt one of the purposes to have no usage of kingpin in the individual collectors to prevent possible kingpin overlap?

Reply:
> Each config needs to pass 3 things: name, default and helptext. 

IMO only the default is relevant here, but isn't strictly necessary for using as a library. 

> Also isnt one of the purposes to have no usage of kingpin in the individual collectors to prevent possible kingpin overlap?

Yes, but the method to register flags should accept a kingpin application as an argument to avoid this problem. (You can also just avoid calling the register flags method entirely from the library, like we would do, since we'll be unmarshaling config from YAML instead)